features/networking/multus.feature

```diff
@@ -1056,3 +1056,113 @@ Feature: Multus-CNI related scenarios
     Then the output should contain:
       | default via 22.2.2.254 dev net1  |
 
+  # @author anusaxen@redhat.com
+  # @case_id OCP-22504
+  @admin
+  Scenario: The multus adminssion controller should be able to detect that the pod is using net-attach-def in other namespaces when the isolation is enabled
+    Given I create 2 new projects
+    # Create the net-attach-def via cluster admin
+    When I run oc create as admin over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/macvlan-bridge.yaml" replacing paths:
+      | ["metadata"]["name"]      | macvlan-bridge-25657    |
+      | ["metadata"]["namespace"] | <%= project(-1).name %> |    
+    Then the step should succeed
+    Given I use the "<%= project(-2).name %>" project
+    # Create a pod in new project consuming net-attach-def from 1st project
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/1interface-macvlan-bridge.yaml" replacing paths:
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | macvlan-bridge-25657 |
+    Then the step should succeed
```

Comment:
I think the pod should NOT be created if no related 'net-attach-def' in the current project. the admission controller should block  pod to be created and given the correct error message return. if not, I think it's a bug @weliang1

Reply:
@zhaozhanqi @weliang1 This functionally works as expected. The step should definitely succeed which means that pod should be created and yaml file is taken successfully. As we know that  pod creation doesn't necessarily mean in Ready state. That what happening, In upcoming steps we are checking that pod with "net-attach-def" warning. I have also added comments and made sure too that pod reflects "ContainerCreating" state for readers understanding. WDYT?

Reply:
hmm, I do not think this is expected.  why multus adminssion controller should block the pod created. eg. see sriov admission
```
oc create -f intel-netdevice.yaml -n z2
Error from server: error when creating "intel-netdevice.yaml": admission webhook "network-resources-injector-config.k8s.io" denied the request: could not find network attachment definition 'z2/intel-netdevice-rhcos': could not get Network Attachment Definition z2/intel-netdevice-rhcos: the server could not find the requested resource
```

Reply:
Let me confirm with dev regarding appropriate behavior

Reply:
Seems like its working as designed currently as per discussion on multus forum https://bugzilla.redhat.com/show_bug.cgi?id=1754512. But they might see this in future.